compiler/rustc_lint/src/types.rs

```diff
@@ -1117,6 +1117,11 @@ impl<'a, 'tcx> ImproperCTypesVisitor<'a, 'tcx> {
                 FfiSafe
             }
 
+            ty::TyAlias(def_id, _) => {
```

Comment:
why are we throwing away the substs here?

Reply:
What should I use it for? The only case I can see where it's used is for enum variant and field.

Reply:
I think we need to check the substituted type for FFI safety

`type Foo<T> = *const T`

when checking that `Foo<i32>` is safe for ffi, we need to check that `*const i32` is safe for FFI, not `*const T`

Reply:
I added check for all substs as well.